mirror-service/src/main/java/org/hyades/nvd/NvdToCyclonedxParser.java

```diff
@@ -0,0 +1,225 @@
+package org.hyades.nvd;
+
+import io.github.jeremylong.nvdlib.nvd.Config;
+import io.github.jeremylong.nvdlib.nvd.CpeMatch;
+import io.github.jeremylong.nvdlib.nvd.CveItem;
+import io.github.jeremylong.nvdlib.nvd.CvssV2;
+import io.github.jeremylong.nvdlib.nvd.CvssV20;
+import io.github.jeremylong.nvdlib.nvd.CvssV30;
+import io.github.jeremylong.nvdlib.nvd.CvssV30__1;
+import io.github.jeremylong.nvdlib.nvd.CvssV31;
+import io.github.jeremylong.nvdlib.nvd.CvssV31__1;
+import io.github.jeremylong.nvdlib.nvd.LangString;
+import io.github.jeremylong.nvdlib.nvd.Metrics;
+import io.github.jeremylong.nvdlib.nvd.Node;
+import io.github.jeremylong.nvdlib.nvd.Reference;
+import io.github.jeremylong.nvdlib.nvd.Weakness;
+import org.apache.commons.lang3.StringUtils;
+import org.cyclonedx.model.Bom;
+import org.cyclonedx.model.Component;
+import org.cyclonedx.model.ExternalReference;
+import org.cyclonedx.model.vulnerability.Vulnerability;
+import org.hyades.resolver.CweResolver;
+
+import java.sql.Date;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Parser and processor of NVD data feeds.
+ */
+public final class NvdToCyclonedxParser {
+
+    private final Bom cdxBom = new Bom();
+    private final Vulnerability cdxVuln = new Vulnerability();
```

Comment:
Using fields for these variables means that instances of `NvdToCycloneDxParser` can't safely be reused. If these variables would be created in `parse` instead, `parse` could even be made `static`.

Reply:
Yes, but in that case, there are two extracted methods `getBomRef` and `parseCpe` which are setting components in the box and returning other values. These would need to be merged back inside `parse` method.

Reply:
Or I can return a map of bov and values from these methods to `parse`.